docs/v1/P4Runtime-Spec.mdk

```diff
@@ -782,67 +784,146 @@ scheme in an out-of-band fashion.
+   `role.config` for the given `role.id`.  Furthermore:
+
+    1. If there was no master for this `device_id` and `role_id` before and
+       there are no `Write` requests still processing from a previous master,
+       then the server immediately sends an advisory notification to all
+       controllers for this `device_id` and `role_id`. See the
+       [following section](#sec-mastership-notification) for the format of the
+       advisory message.
+
+    2. If there was a previous master or `Write` requests in flight, then the
+       server carries out the following steps (in this order):
+
+        1. The server stops accepting `Write` requests from the previous master
+           (if there is one). At this point, the server will reject all `Write`
+           requests with `PERMISSION_DENIED`.
+
+        2. The server notifies all controllers other than the new master of the
+           mastership change by sending the advisory notification described in
+           the [following section](#sec-mastership-notification).
+
+        3. The server will finish processing any `Write` requests that have
+           already started. If there are errors, they are reported as usual to
+           the previous master. If the previous master has already disconnected,
+           any possible errors are dropped and not reported.
+
+        4. The server now accepts the current controller as the new master, thus
+           accepting `Write` requests from this controller. The server updates
+           the highest election ID (&ie; `election_id_past`) it has seen for
+           this `device_id` and `role_id` to `election_id`.
+
+        5. The server notifies the new master by sending the advisory message
+           described in the [following section](#sec-mastership-notification).
+
+2. Otherwise, the controller becomes a slave: If the controller was previously a
+   master (and downgraded itself), then an advisory message is sent to all
+   controllers for this `device_id` and `role_id`.  Otherwise, the advisory
+   message is only sent to the controller that sent the initial
+   `MasterArbitrationUpdate`.  See the
+   [following section](#sec-mastership-notification) for the format of the
+   advisory message.
+
+## Mastership Notifications { #sec-mastership-notification}
+
+For any given `device_id` and `role_id`, any time a new master is chosen, a
+master downgrades its status to a slave, a master disconnects, or the
+`role.config` is updated by the master, all controllers for that
+(`device_id`, `role_id`) are informed of this by sending a
+`StreamMessageResponse`. The `MasterArbitrationUpdate` is populated as follows:
+
+* `device_id` and `role.id` as given.
+
+* `role.config` is set to the role configuration the server received most
+  recently in a `MasterArbitrationUpdate` from a master.
+
+* If there currently is a master, then `election_id` is set to the `election_id`
```

Comment:
I believe this point could be simplified by removing the case "If there currently is a master...". Indeed, election_id should always be set to the highest election ID that the server has seen, which includes the case where a master exists.

Also, I would enforce the case when there has not been any master at all by removing the parentheses. Maybe you can create a new bullet point:

If there has not been any master at all, the `election_id` is left unset.

Reply:
Done.

Thank you for your review, I applied all changes.